src/graphcast_agent/message_typing.rs

```diff
@@ -261,6 +261,15 @@ impl<T: Message + ethers::types::transaction::eip712::Eip712 + Default + Clone +
 
     /// Recover sender address from Graphcast message radio payload
     pub fn recover_sender_address(&self) -> Result<String, BuildMessageError> {
+        // These mocks are only used in tests and are temporary
+        // Will be removed once tests are ported to use cfg(test)
+        if env::var("CHECK").is_ok() {
```

Comment:
If the mock message is filled correctly, should the mocked address be recovered just fine without this conditional?